src/main/java/org/mifos/processor/bulk/zeebe/worker/OrderingWorker.java

```diff
@@ -26,26 +32,85 @@ public void setup() {
         newWorker(Worker.ORDERING, (client, job) -> {
             logger.debug("Job '{}' started from process '{}' with key {}", job.getType(), job.getBpmnProcessId(), job.getKey());
             Map<String, Object> variables = job.getVariablesAsMap();
+            Exchange exchange = new DefaultExchange(camelContext);
+
             if (workerConfig.isOrderingWorkerEnabled) {
                 variables.put(ORDERING_FAILED, false);
+                String filename = (String) variables.get(FILE_NAME);
+                exchange.setProperty(SERVER_FILE_NAME, filename);
+
+                try {
+                    sendToCamelRoute(RouteId.ORDERING, exchange);
+                    assert !exchange.getProperty(ORDERING_FAILED, Boolean.class);
+                } catch (Exception e) {
+                    variables.put(ORDERING_FAILED, true);
+                }
+                variables.put(ORDERING_FAILED, false);
+                variables.put(ORDERED_BY, exchange.getProperty(ORDERED_BY));
             }
 
-            String filename = (String) variables.get(FILE_NAME);
-            Exchange exchange = new DefaultExchange(camelContext);
-            exchange.setProperty(SERVER_FILE_NAME, filename);
+            if(workerConfig.isTransactionDeduplicationEnabled){
+                List<Transaction> transactionList = exchange.getProperty(TRANSACTION_LIST, List.class);
+                removeDuplicates(transactionList, workerConfig.isOrderingWorkerEnabled);
+                variables.put(TRANSACTION_LIST, transactionList);
+            }
+            client.newCompleteCommand(job.getKey()).variables(variables).send();
+        });
+    }
+
+    private void removeDuplicates(List<Transaction> transactionList, boolean orderingEnabled){
+        if(orderingEnabled){
+            removeDuplicatesIfOrderingEnabled(transactionList);
+            return;
+        }
+        removeDuplicatesIfOrderingDisabled(transactionList);
+    }
+
+    private void removeDuplicatesIfOrderingEnabled(List<Transaction> transactionList){
 
-            try {
-                sendToCamelRoute(RouteId.ORDERING, exchange);
-                assert !exchange.getProperty(ORDERING_FAILED, Boolean.class);
-            } catch (Exception e) {
-                variables.put(ORDERING_FAILED, true);
+        for (int i = 0; i <transactionList.size()-1; i++) {
+            Transaction currentTransaction = transactionList.get(i);
+            Transaction nextTransaction = transactionList.get(i + 1);
+
+            if (currentTransaction == null || nextTransaction == null) {
+                continue;
+            }
+            String currentPayeeDetail = fetchPayeeDetail(currentTransaction);
+            String nextPayeeDetail = fetchPayeeDetail(nextTransaction);
+
+            if (currentPayeeDetail.equals(nextPayeeDetail)) {
```

Comment:
not null safe , this method: fetchPayeeDetail(currentTransaction) can possibly return null ( as the method has a return null )